sentry_sdk/utils.py

```diff
@@ -891,21 +891,32 @@ def __init__(self, waiting_time, configured_timeout):
         threading.Thread.__init__(self)
         self.waiting_time = waiting_time
         self.configured_timeout = configured_timeout
+        self.stop_thread = False
 
     def run(self):
         # type: () -> None
 
-        time.sleep(self.waiting_time)
+        raise_exception = True
+        threading_is_running = True
+        while threading_is_running and self.waiting_time > 1:
+            time.sleep(1)
+            self.waiting_time = self.waiting_time - 1
+            if self.stop_thread:
+                raise_exception = False
+                threading_is_running = False
 
-        integer_configured_timeout = int(self.configured_timeout)
+        if raise_exception:
```

Comment:
This is racy -- there is an execution flow where `raise_exception` is `True` and `self.stop_thread` is also `True` (think the main thread may change `stop_thread` between the end of the `while` loop and the beginning of the `if` block).